src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooService.java

```diff
@@ -584,22 +584,30 @@ protected void addFeedbackToResult(Result result, AbstractBuildResultNotificatio
         final var programmingLanguage = programmingExercise.getProgrammingLanguage();
         final var projectType = programmingExercise.getProjectType();
 
+        result.setTestCaseCount(0);
+        result.setPassedTestCaseCount(0);
+        result.setCodeIssueCount(0);
+
         for (final var job : jobs) {
             // 1) add feedback for failed test cases
             for (final var failedTest : job.getFailedTests()) {
                 result.addFeedback(feedbackRepository.createFeedbackFromTestCase(failedTest.getName(), failedTest.getErrors(), false, programmingLanguage, projectType));
             }
+            result.setTestCaseCount(result.getTestCaseCount() + job.getFailedTests().size());
```

Comment:
Optionally: A method `addTestCaseCount` might be helpful.

Reply:
I'd keep the code rather short and don't add methods if they are not used too often or the code is too complicated